### PR TITLE
feat: allow multiple components with elementConstantAttributeBuilder

### PR DESCRIPTION
**Pull request by jamesmporter**
Addresses: #1404

**Example usage:**
```js
    const xYValueAttribute = elementConstantAttributeBuilder()
        .size(2)
        .value((data, element, component) => data[element][component]);
```

### CODE REVIEW

**Review by chrisprice — CHANGES_REQUESTED** [review]
As discussed please can we remove the `component` argument from the `value` function and return a tuple of component values instead.

File: packages/d3fc-webgl/src/buffers/elementConstantAttributeBuilder.js [hunk 1]
```diff
@@ -6,23 +6,33 @@ export default () => {
...
+        if (componentCount > 1) {
+            for (let element = 0; element < elementCount; element++) {
+                for (
+                    let component = 0;
+                    component < componentCount;
+                    component++
+                ) {
+                    projectedData[
+                        offset + element * componentCount + component
+                    ] = value(data, element)[component];
```
Comment (chrisprice) [inline packages/d3fc-webgl/src/buffers/elementConstantAttributeBuilder.js:27]: Won't this invoke the function `componentCount` times more than required?

**Review by chrisprice — CHANGES_REQUESTED** [review]

File: packages/d3fc-webgl/src/buffers/elementConstantAttributeBuilder.js (hunk 1 above)
Comment (OliverForeman) [reply packages/d3fc-webgl/src/buffers/elementConstantAttributeBuilder.js:27]: This is now addressed in #1436

**Comment by OliverForeman** [comment]
This is now duplicated by #1436

**Pull request closed without merging**